parquet-thrift/src/main/java/parquet/thrift/ThriftSchemaConverter.java

```diff
@@ -93,7 +94,7 @@ private StructType toStructType(TStructDescriptor struct) {
                         Requirement.fromType(field.getFieldMetaData().requirementType);
         children.add(toThriftField(field.getName(), field, req));
       }
-      return new StructType(children);
+      return new StructType(children, TBaseUtil.isUnion(struct.getThriftClass()));
```

Comment:
Yeah, I only made this thinking the detection was going to be more involved. I'll just move it here.